librz/debug/p/native/linux/linux_debug.c

```diff
@@ -872,11 +872,59 @@ RzList /*<RzDebugPid *>*/ *linux_pid_list(int pid, RzList /*<RzDebugPid *>*/ *li
 	return list;
 }
 
+static int thread_find(int *tid, RzDebugPid *t, void *user) {
+	return (t && (t->pid == *tid)) ? 0 : 1;
+}
+
+RZ_API RZ_OWN RzDebugPid /*<RzDebugPid *>*/ *rz_debug_get_thread(RzList *th_list, int tid) {
+	if (!th_list) {
+		return NULL;
+	}
+
+	RzListIter *it = rz_list_find(th_list, (const void *)(size_t)&tid,
+		(RzListComparator)&thread_find, NULL);
+	if (!it) {
+		return NULL;
+	}
+	return (RzDebugPid *)rz_list_iter_get_data(it);
+}
+
+RZ_API RZ_OWN ut64 get_linux_tls_val(RZ_NONNULL RzDebug *dbg, int tid) {
```

Comment:
No need for `RZ_OWN` here since it's not a pointer/memory value.

Reply:
If you add `RZ_NONNULL` you also need to add assert `rz_return_val_if_fail()` with that argument.